File: distinctIntegerSubArray.cpp
int Solution::solve(vector<int> &A, int B) {
    int result = 0, temp = 0, count = 0, j = 0, n = A.size();
    unordered_map<int, int> countArray;
    for(int i = 0; i < n; ++i) {
        if (countArray[A[i]]++ == 0) 
            ++count;
        if (count > B){
            temp = 0;
            --count;
            --countArray[A[j++]];
        }
        while (countArray[A[j]] > 1){
            ++temp;
            --countArray[A[j++]];
        }
        if (count == B) 
            result = result + temp + 1;
    }
    return result;
}

File: distinctIntegerSubArray.py
from collections import Counter
class Solution:
    # @param A : list of integers
    # @param B : integer
    # @return an integer
    def solve(self, A, B):
        n, res, count, temp = len(A), 0, 0, 0
        countArray, j = Counter(A), 0
        for i in range(n):
            if countArray[A[i]] == 1:
                countArray[A[i]] += 1
                count += 1
            if count > B:
                j += 1
                countArray[A[j]] -= 1
                count -= 1
                temp = 0
            while countArray[A[j]] > 1:
                temp += 1
                j += 1
                countArray[A[j]] -= 1
            if count == B:
                res += temp + 1
        return res